src/core_plugins/kibana/public/store.js

```diff
@@ -0,0 +1,12 @@
+import { createStore, applyMiddleware } from 'redux';
```

Comment:
Because all kibana core apps are really part of a single plugin app, I created the store in a place that will be available to all of them, not just dashboard specific.

Reply:
I'm not really for or against this idea, but is that the plan then? Each "app" will get one universal store?

Reply:
Also, any consider for a more granular state layout? Having just a single top-level collection of reducers could result in gigantic and possibly hard to understand and work with collection of reducers. Maybe the intention is that we simply break things up as they grow?

Reply:
> I'm not really for or against this idea, but is that the plan then? Each "app" will get one universal store?
  
It was my idea, but I'm open to others as well.  My reasoning is that with redux there is a single store, but each plugin is separate and so it will also have it's own redux store (or not, if it's in angular, or something else).  Then they will all communicate via the new platform/build system (eventually).

> Also, any consider for a more granular state layout? Having just a single top-level collection of reducers could result in gigantic and possibly hard to understand and work with collection of reducers. Maybe the intention is that we simply break things up as they grow

I was thinking it'd look like:
```
export const store = createStore(
  reducers,
  {
    dashboard: getInitialDashboardState(),
    visualization: getInitialVisualizationState(),
    discover: getInitialDiscoverState(),
  },
  applyMiddleware(thunkMiddleware)
);
```

But I could actually see some of that being taken out and placed globally - e.g. filters and the time picker which is relevant to all the core plugins.

All of this is flexible though.  Doing it this way gives other apps the ability to add separate state info into the store, but also lets us re-organize later if need be.

Reply:
I think time picker and other state all still fits this pretty well honestly. Especially if we use selectors, then the shape kind of doesn't matter. 

My concern is more about the size of reducers I guess. Even considering just Dashboard, there could be a lot of actions and reducers there, and putting everything in a single file starts to get pretty hard to deal with. It's all just javascript, so we could push the concerns of breaking it down further into each of these app-level reducers, or we could just tweak stuff as we go in this root reducer definition. 

Either way is fine, and starting off the way you have it is fine too, just curious what this might look like as things grow.

Reply:
Ah yea, I think I'll end up breaking up the reducers into separate files.  For right now, they seemed small enough where only one really has any logic, that putting them in a file was sufficient.  But as the angular lessens and react grows, I imagine it will make more sense to split it out.

Reply:
I think I might actually break some of it up now, as I'm switching to using the handleActions and createAction and following some of the patterns here https://github.com/bmcconaghy/x-pack-kibana/tree/license_management/plugins/license_management/public/store